src/main/java/io/vertx/core/spi/metrics/NetworkMetrics.java

```diff
@@ -20,7 +20,12 @@
 
 /**
  * An SPI used internally by Vert.x to gather metrics on a net socket which serves
- * as a base class for TCP or UDP.
+ * as a base class for TCP or UDP.<p/>
+ *
+ * All the methods on this object including the methods inherited from the super interfaces are invoked
+ * with {@link io.vertx.core.Context} and thread of the network server/client and therefore are the same than
```

Comment:
so
- the SPI can determine at creation time if it's in a event loop verticle / worker verticle
- when it's a worker verticle, what is guaranteed is:
    - the life cycle of a request will always use the same thread : so any object created during the life cycle of the request will use the same thread
    - at most one thread will make a callback in the spi

therefore I think the super interface like TCPMetrics / NetworkMetrics should only specify to look at the doc of the _concrete_ interfaces and these should be updated with this info.

does it make sense ?